api/v1alpha1/validation.go

```diff
@@ -485,3 +538,82 @@ func getAppName(app ApplicationSpec, appType ApplicationProviderType) (string, e
 		return "", fmt.Errorf("unsupported application provider type: %s", appType)
 	}
 }
+
+// note: this regex was taken from the github.com/kubernetes/kubernetes/pkg/apis/batch/validation/validation.go
+// https://data.iana.org/time-zones/theory.html#naming
+// * A name must not be empty, or contain '//', or start or end with '/'.
+// * Do not use the file name components '.' and '..'.
+// * Within a file name component, use only ASCII letters, '.', '-' and '_'.
+// * Do not use digits, as that might create an ambiguity with POSIX TZ strings.
+// * A file name component must not exceed 14 characters or start with '-'
+//
+// 0-9 and + characters are tolerated to accommodate legacy compatibility names
+var validTimeZoneCharacters = regexp.MustCompile(`^[A-Za-z\.\-_0-9+]{1,14}$`)
+
+// validateTimeZone validates the time zone string. it must be a valid IANA time zone identifier.
+func validateTimeZone(timeZone string) []error {
```

Comment:
We're keeping most of the reusable, API version-independent validation code in https://github.com/flightctl/flightctl/tree/main/internal/util/validation, but we can move that later if needed.